sympy/solvers/ode.py

```diff
@@ -1001,7 +995,6 @@ class in it.  Note that a hint may do this anyway if
     r3 = {'xi': xi, 'eta': eta}  # Used for the lie_group hint
     boundary = {}  # Used to extract initial conditions
     C1 = Symbol("C1")
-    eq = expand(eq)
```

Comment:
Rather than removing this, how about adding `eq_orig = eq` on the line above?

Reply:
I think nth_algebraic could also benefit from using the original equation where possible

Reply:
I am removing it because due to the expansion structure of the equation is changing. 
But I am using the equation similar to the original for others type of matching except factorable.

Reply:
I think it would be better to do `eq_orig = eq` so that we aren't affecting anything else

Reply:
Or is it because this is in a different function. Maybe it is better to leave it like this

Reply:
I am leaving the original equation as it is.